src/rdkafka_admin.c

```diff
@@ -7049,6 +7099,51 @@ const rd_kafka_error_t **rd_kafka_ListConsumerGroups_result_errors(
  *
  */
 
+/**
+ * @brief Parse authorized_operations returned in
+ * - DescribeConsumerGroups
+ * - DescribeTopics
+ * - DescribeCluster
+ *
+ * @param authorized_operations returned by RPC, containing operations encoded
+ *                              per-bit.
+ * @param cntp is set to the count of the operations.
+ * @returns rd_kafka_AclOperation_t *
+ */
+static rd_kafka_AclOperation_t *
+rd_kafka_AuthorizedOperations_parse(int32_t authorized_operations,
+                                    size_t *cntp) {
+        rd_kafka_AclOperation_t i;
+        int j     = 0;
+        int count = 0;
+        rd_kafka_AclOperation_t *operations;
+
+        /* In case of authorized_operations not requested, return NULL. */
+        if (authorized_operations < 0) {
+                *cntp = 0;
+                return NULL;
+        }
+
+        /* Count number of bits set. ALL, ANY and UNKNOWN bits are skipped as
+         * they are always unset as per KIP-430. */
+        for (i = RD_KAFKA_ACL_OPERATION_READ; i < RD_KAFKA_ACL_OPERATION__CNT;
+             i++)
+                count += ((authorized_operations >> i) & 1);
+
+        j          = 0;
+        operations = rd_malloc(sizeof(rd_kafka_AclOperation_t) * count);
```

Comment:
Count can be zero and in that case `malloc(0)` is implementation defined.
We can `rd_malloc(1) `if it's zero while keeping count as 0, so we have a pointer instead of implementation defined behavior. Or we can set count to 0 if it's empty or to -1 if it was NULL.